python-package/lightgbm/sklearn.py

```diff
@@ -14,6 +14,17 @@
                      pd_DataFrame)
 from .engine import train
 
+try:
+    from joblib import cpu_count
+    cpu_count_lib = "joblib"
+except ImportError:
+    try:
+        from psutil import cpu_count
+        cpu_count_lib = "psutil"
+    except ImportError:
+        from multiprocessing import cpu_count
+        cpu_count_lib = "multiprocessing"
```

Comment:
Please move all this helper stuff to the [`compat.py`](https://github.com/microsoft/LightGBM/blob/master/python-package/lightgbm/compat.py) file with the following encapsulation:
```python
"""cpu_count()"""
try:
    from joblib import cpu_count

    def _LGBMCpuCount():
        return cpu_count(only_physical_cores=True)
except ImportError:
    try:
        from psutil import cpu_count

        def _LGBMCpuCount():
            return cpu_count(logical=False)
    except ImportError:
        from multiprocessing import cpu_count

        def _LGBMCpuCount():
            return cpu_count()
```